[Spanner] Error decoding message: Cannot convert '' to string in RequestOptions

OS: alpine 3.14
PHP version: 7.4.24
Package name and version:
google/cloud-spanner v1.46.0
grpc v1.41.0

When I run `$database->runTransaction(...)` I get the following error.

```
Google\ApiCore\ValidationException: Error decoding message: Cannot convert '' to string

/project/vendor/google/gax/src/Serializer.php:128
/project/vendor/google/cloud-spanner/src/Connection/Grpc.php:913
/project/vendor/google/cloud-spanner/src/Operation.php:751
/project/vendor/google/cloud-spanner/src/Operation.php:436
/project/vendor/google/cloud-spanner/src/Database.php:856
/project/vendor/google/cloud-spanner/src/Database.php:873
/project/vendor/google/cloud-core/src/Retry.php:78
/project/vendor/google/cloud-spanner/src/Database.php:897
/project/tests/Colopl/Spanner/ConnectionTest2.php:49

Caused by
Exception: Cannot convert '' to string

/project/vendor/google/cloud-spanner/src/V1/RequestOptions.php:197
/project/vendor/google/gax/src/Serializer.php:420
/project/vendor/google/gax/src/Serializer.php:126
/project/vendor/google/cloud-spanner/src/Connection/Grpc.php:913
/project/vendor/google/cloud-spanner/src/Operation.php:751
/project/vendor/google/cloud-spanner/src/Operation.php:436
/project/vendor/google/cloud-spanner/src/Database.php:856
/project/vendor/google/cloud-spanner/src/Database.php:873
/project/vendor/google/cloud-core/src/Retry.php:78
/project/vendor/google/cloud-spanner/src/Database.php:897
/project/tests/Colopl/Spanner/ConnectionTest2.php:49
```

I'm guessing it was introduced in [v0.168.0](https://github.com/googleapis/google-cloud-php/releases/tag/v0.168.0)

Sorry, I was running the non-test environment on the wrong version (cloud-spanner 1.45).
I tried it again with 1.46 and was able to reproduce the error with the following code.

```
$client = new SpannerClient();
$database = $client->connect('my-instance', 'my-database');
$database->runTransaction(function() {
});
```

```
[2021-10-11 13:29:51.577] [ERROR] Exception: Cannot convert '' to string in /project/vendor/google/cloud-spanner/src/V1/RequestOptions.php:197
Stack trace:
#0 /project/vendor/google/gax/src/Serializer.php(420): Google\Cloud\Spanner\V1\RequestOptions->setTransactionTag(NULL)
#1 /project/vendor/google/gax/src/Serializer.php(126): Google\ApiCore\Serializer->decodeMessageImpl(Object(Google\Cloud\Spanner\V1\RequestOptions), Object(Google\Protobuf\Descriptor), Array)
#2 /project/vendor/google/cloud-spanner/src/Connection/Grpc.php(913): Google\ApiCore\Serializer->decodeMessage(Object(Google\Cloud\Spanner\V1\RequestOptions), Array)
#3 /project/vendor/google/cloud-spanner/src/Operation.php(751): Google\Cloud\Spanner\Connection\Grpc->beginTransaction(Array)
#4 /project/vendor/google/cloud-spanner/src/Operation.php(436): Google\Cloud\Spanner\Operation->beginTransaction(Object(Google\Cloud\Spanner\Session\Session), Array)
#5 /project/vendor/google/cloud-spanner/src/Database.php(856): Google\Cloud\Spanner\Operation->transaction(Object(Google\Cloud\Spanner\Session\Session), Array)
#6 [internal function]: Google\Cloud\Spanner\Database->Google\Cloud\Spanner\{closure}(Object(Google\Cloud\Spanner\Session\Session), Array)
#7 /project/vendor/google/cloud-spanner/src/Database.php(873): call_user_func_array(Object(Closure), Array)
#8 [internal function]: Google\Cloud\Spanner\Database->Google\Cloud\Spanner\{closure}(Object(Closure), Object(Google\Cloud\Spanner\Session\Session), Array)
#9 /project/vendor/google/cloud-core/src/Retry.php(78): call_user_func_array(Object(Closure), Array)
#10 /project/vendor/google/cloud-spanner/src/Database.php(897): Google\Cloud\Core\Retry->execute(Object(Closure), Array)
#11 /project/app/Console/Commands/TestCommand.php(21): Google\Cloud\Spanner\Database->runTransaction(Object(Closure))
#12 /project/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\Console\Commands\TestCommand->handle()

Next Google\ApiCore\ValidationException: Error decoding message: Cannot convert '' to string in /project/vendor/google/gax/src/Serializer.php:128
Stack trace:
#0 /project/vendor/google/cloud-spanner/src/Connection/Grpc.php(913): Google\ApiCore\Serializer->decodeMessage(Object(Google\Cloud\Spanner\V1\RequestOptions), Array)
#1 /project/vendor/google/cloud-spanner/src/Operation.php(751): Google\Cloud\Spanner\Connection\Grpc->beginTransaction(Array)
#2 /project/vendor/google/cloud-spanner/src/Operation.php(436): Google\Cloud\Spanner\Operation->beginTransaction(Object(Google\Cloud\Spanner\Session\Session), Array)
#3 /project/vendor/google/cloud-spanner/src/Database.php(856): Google\Cloud\Spanner\Operation->transaction(Object(Google\Cloud\Spanner\Session\Session), Array)
#4 [internal function]: Google\Cloud\Spanner\Database->Google\Cloud\Spanner\{closure}(Object(Google\Cloud\Spanner\Session\Session), Array)
#5 /project/vendor/google/cloud-spanner/src/Database.php(873): call_user_func_array(Object(Closure), Array)
#6 [internal function]: Google\Cloud\Spanner\Database->Google\Cloud\Spanner\{closure}(Object(Closure), Object(Google\Cloud\Spanner\Session\Session), Array)
#7 /project/vendor/google/cloud-core/src/Retry.php(78): call_user_func_array(Object(Closure), Array)
#8 /project/vendor/google/cloud-spanner/src/Database.php(897): Google\Cloud\Core\Retry->execute(Object(Closure), Array)
#9 /project/app/Console/Commands/TestCommand.php(21): Google\Cloud\Spanner\Database->runTransaction(Object(Closure))
```

Btw, the `google/gax` version we are using is `v1.9.1`.

--- a/Spanner/src/Operation.php
+++ b/Spanner/src/Operation.php
@@ -425,12 +425,15 @@ class Operation
      */
     public function transaction(Session $session, array $options = [])
     {
-        $transactionTag = $this->pluck('tag', $options, false);
         $options += [
             'singleUse' => false,
             'isRetry' => false,
-            'requestOptions' => ['transactionTag' => $transactionTag]
+            'requestOptions' => []
         ];
+        $transactionTag = $this->pluck('tag', $options, false);
+        if (isset($transactionTag)) {
+            $options['requestOptions']['transactionTag'] = $transactionTag;
+        }
 
         if (!$options['singleUse']) {
             $res = $this->beginTransaction($session, $options);

--- a/Spanner/tests/Unit/OperationTest.php
+++ b/Spanner/tests/Unit/OperationTest.php
@@ -304,6 +304,23 @@ class OperationTest extends TestCase
         $this->assertEquals(self::TRANSACTION, $t->id());
     }
 
+    public function testTransactionNoTag()
+    {
+        $this->connection->beginTransaction(Argument::allOf(
+            Argument::withEntry('database', self::DATABASE),
+            Argument::withEntry('session', $this->session->name()),
+            Argument::withEntry('requestOptions', [])
+        ))
+            ->shouldBeCalled()
+            ->willReturn(['id' => self::TRANSACTION]);
+
+        $this->operation->___setProperty('connection', $this->connection->reveal());
+
+        $t = $this->operation->transaction($this->session);
+        $this->assertInstanceOf(Transaction::class, $t);
+        $this->assertEquals(self::TRANSACTION, $t->id());
+    }
+
     public function testSnapshot()
     {
         $this->connection->beginTransaction(Argument::allOf(

